core/src/main/scala/zio/jdbc/ZConnectionPool.scala

```diff
@@ -167,12 +167,16 @@ object ZConnectionPool {
                     for {
                       connection <- pool.get
                       _          <- ZIO.addFinalizerExit {
-                                      case Exit.Success(_) => ZIO.unit
-                                      case Exit.Failure(_) => connection.rollback.ignoreLogged
+                                      case Exit.Success(_) =>
+                                        ZIO.unit
+                                      case Exit.Failure(_) =>
+                                        for {
+                                          autoCommitMode <- connection.access(_.getAutoCommit).orElseSucceed(true)
```

Comment:
We have to use `.orElseSucceed` because it is not allowed to throw exceptions inside `.addFinalizerExit`.